### ISSUE TITLE
OCommandExecutorNotFoundException when executing a scheduled function

### ISSUE THREAD

**Issue by johny65** [orientechnologies/orientdb#9416]
## Expected behavior  

Create a function using SQL and schedule it to fire with the event scheduler, and the function is correctly executed when it's time.

## Actual behavior  

If the function contains syntax of batch SQL, the execution fails with exception `com.orientechnologies.orient.core.command.OCommandExecutorNotFoundException`. I need to schedule a SQL function specifically using the `foreach` keyword but when the scheduler has to execute it, it fails.

## Steps to reproduce  

This is a simplified example. Create a function like this:

```sql
CREATE FUNCTION testFunc
"let $ee = select from E where endDate < sysdate();
foreach ($e in $ee) {
    /* do something with $e */
}"
LANGUAGE SQL;
```

In Studio, in the "Functions" section if you execute it, it runs ok: "Function testFunc executed in 0.053 sec.".

Now create an event:

```sql
INSERT INTO OSchedule SET name = 'testEvent',
    function = (SELECT FROM OFunction WHERE name = 'testFunc'),
    rule = "0 0 8 1/1 * ? *";
```

When OrientDB executes the event, the server logs:

```
Scheduled event 'testEvent' executionId=1 completed with result: null [OScheduledEvent]$ANSI{green {db=testdb}} Error during execution of scheduled function
com.orientechnologies.orient.core.command.OCommandExecutorNotFoundException: Cannot find a command executor for the command request: sql.FOREACH ($e IN $ee) {
	DB name="testdb"
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.parse(OCommandExecutorSQLDelegate.java:54)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.parse(OCommandExecutorSQLDelegate.java:39)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:4407)
	at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:68)
	at com.orientechnologies.orient.core.command.script.OCommandExecutorScript.executeCommand(OCommandExecutorScript.java:517)
	at com.orientechnologies.orient.core.command.script.OCommandExecutorScript.executeSQLScript(OCommandExecutorScript.java:382)
	at com.orientechnologies.orient.core.command.script.OCommandExecutorScript.executeSQL(OCommandExecutorScript.java:228)
	at com.orientechnologies.orient.core.command.script.OCommandExecutorScript.executeInContext(OCommandExecutorScript.java:104)
	at com.orientechnologies.orient.core.command.script.OCommandExecutorScript.execute(OCommandExecutorScript.java:88)
	at com.orientechnologies.orient.core.metadata.function.OFunction.execute(OFunction.java:206)
	at com.orientechnologies.orient.core.schedule.OScheduledEvent.executeFunction(OScheduledEvent.java:258)
	at com.orientechnologies.orient.core.schedule.OScheduledEvent.access$200(OScheduledEvent.java:47)
	at com.orientechnologies.orient.core.schedule.OScheduledEvent$ScheduledTimer.run(OScheduledEvent.java:81)
	at com.orientechnologies.orient.core.Orient$4.run(Orient.java:452)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:556)
	at java.base/java.util.TimerThread.run(Timer.java:506)
```

**Comment by lvca** [comment]
I believe the issue here is that the SQL script command executor should be used. Investigating...

**Comment by lvca** [comment]
`FOREACH` can only be used in the console, it's not part of OrientDB's SQL language. @luigidellaquila any plan to include it in SQL?

**Comment by johny65** [comment]
Hi @lvca. I'm using `FOREACH` in some queries without problem, through Java (the next query is just an example, it could be done without the for loop):

```java
String query = "let $ee = select from E where endDate is null; foreach ($e in $ee) { update $e set endDate = sysdate(); }";
ODatabaseSession odb = ...
odb.execute("sql", query);
```

The previous code works without problem.

**Comment by lvca** [comment]
@johny65 you're right, I found FOREACH in the SQL definition, but its management it's missing in the class `ODefaultCommandExecutorSQLFactory`. Working on it

**Comment by lvca** [comment]
Just fixed on both 3.1.x and develop (3.2)
